lib/chainer/functions/connection/embed_id.rb

```diff
@@ -10,36 +10,38 @@ def self.embed_id(x, w, ignore_label: nil)
           self.new(ignore_label: ignore_label).(x, w)
         end
 
-        def forward_cpu(inputs)
+        def forward(inputs)
+          xm = Chainer.get_array_module(*inputs)
           (x, w) = inputs
 
           unless @ignore_label
-            return [Chainer::Utils::Array.take(w, x.to_a, axis: 0)]
+            return [Chainer::Utils::Array.take(w, x, axis: 0)]
           end
 
-          mask = Numo::Bit.cast(x.flatten.to_a.map{|i| i == @ignore_label}).reshape(*x.shape)
+          mask = xm::Bit.cast(x.flatten.to_a.map{|i| i == @ignore_label}).reshape(*x.shape)
           if mask.where.empty?
-            return [Chainer::Utils::Array.take(w, x.to_a, axis: 0)]
+            return [Chainer::Utils::Array.take(w, x, axis: 0)]
           end
           x = x.dup
           x[mask.where] = 0
```

Comment:
Also, not sure, but maybe following codes are faster?

```
x_valid = x.ne(@ignore_label)
if x_valid.count > 0
   ...
end
x = x.dup
x *= x_valid
```

Reply:
`x.ne(@ignore_label).count > 0` is not equivalent to `x.eq(@ignore_label).count == 0`

`x.ne(@ignore_label).count == x.size` is equivalent to it. How is it?

Reply:
It looks good to me